Source/common/MOLCodesignChecker.mm

```diff
@@ -77,12 +82,18 @@ - (instancetype)initWithSecStaticCodeRef:(SecStaticCodeRef)codeRef error:(NSErro
   self = [super init];
 
   if (self) {
-    OSStatus status = errSecSuccess;
-    CFErrorRef cfError = NULL;
+    auto [status, scopedError] = ScopedCFError::AssumeFrom(^OSStatus(CFErrorRef *out) {
+      if (CFGetTypeID(codeRef) == SecStaticCodeGetTypeID()) {
+        return SecStaticCodeCheckValidityWithErrors(codeRef, kStaticSigningFlags, NULL, out);
+      } else if (CFGetTypeID(codeRef) == SecCodeGetTypeID()) {
+        return SecCodeCheckValidityWithErrors((SecCodeRef)codeRef, kSigningFlags, NULL, out);
+      } else {
+        return errSecSuccess;
```

Comment:
Not sure success is the right response here, the codeRef should not be empty and so it should only be one of those two.

Perhaps this if/else if/else should be a switch on the CFGetTypeID response and the default should be to return errSecUnimplemented or errSecBadReq